sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingInterpreter.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.scripting
+
+import org.apache.spark.sql.catalyst.analysis.UnresolvedIdentifier
+import org.apache.spark.sql.catalyst.parser.{CompoundBody, CompoundPlanStatement, SingleStatement}
+import org.apache.spark.sql.catalyst.plans.logical.{CreateVariable, DropVariable, LogicalPlan}
+import org.apache.spark.sql.catalyst.trees.Origin
+
+/**
+ * Trait for SQL Scripting interpreters.
+ */
+trait ProceduralLanguageInterpreter {
+  /**
+   * Build execution plan and return statements that need to be executed,
+   *  wrapped in the execution node.
+   * @param compound CompoundBody for which to build the plan.
+   * @return Iterator through collection of statements to be executed.
+   */
+  def buildExecutionPlan(compound: CompoundBody) : Iterator[CompoundStatementExec]
+}
+
+/**
+ * Concrete implementation of the interpreter for SQL scripting.
+ */
+case class SqlScriptingInterpreter() extends ProceduralLanguageInterpreter {
```

Comment:
where do we call `reset`? or it's preparing for loop in the future?

Reply:
You are right, it is a preparation for loops in the future.